src/graphics/vulkan/vulkan_texture.cpp

```
@@ -197,6 +198,68 @@ namespace moth_graphics::graphics::vulkan {
+        region.srcOffset = { static_cast<int32_t>(sourceRect.topLeft.x), static_cast<int32_t>(sourceRect.topLeft.y), 0 };
+        region.extent.width = targetWidth;
+        region.extent.height = targetHeight;
+        region.extent.depth = 1;
+        vkCmdCopyImage(commandBuffer->GetVkCommandBuffer(), m_vkImage, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+                       stagingImage->GetVkImage(), VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, 1, &region);
+
+        commandBuffer->TransitionImageLayout(*stagingImage, targetFormat, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, VK_IMAGE_LAYOUT_GENERAL);
+        commandBuffer->TransitionImageLayout(*this, m_vkFormat, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL, srcLayout);
+        commandBuffer->SubmitAndWait();
+
+        VkImageSubresource subresource{};
```

Clang (14.0.6)

[error] variable 'subresource' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)